### ISSUE TITLE
Implement ML-Based Cross-Sectional Alpha Case Study (XGBoost Sleeve Integration)

### ISSUE THREAD

**Issue by christophermoverton** [christophermoverton/stratlake-trade-engine#179]
# 🧾 **Issue Title**

**Implement ML-Based Cross-Sectional Alpha Case Study (XGBoost Sleeve Integration)**

---

# 🏷️ **Labels**

* enhancement
* research
* alpha-layer
* machine-learning
* milestone-14

---

# 📄 **Description**

## **Context**

The current StratLake alpha layer supports deterministic, feature-driven alphas (e.g., `rank_composite_momentum_2026_q1`) and a fully integrated pipeline:

```plaintext
Features → Alpha → Predict → Align → Evaluate → Map → Sleeve → Portfolio → Compare → Review
```

A real-data case study (`backfilled_2026_q1_alpha_workflow.md`) validates this pipeline using a rule-based alpha. However, the system has not yet been exercised with a **trained machine learning model**.

---

## **Objective**

Introduce a **machine learning-based alpha (XGBoost or LightGBM)** that:

* learns from `features_daily`
* predicts `target_ret_5d`
* produces deterministic outputs
* integrates seamlessly into:

  * alpha evaluation (IC / ICIR)
  * signal mapping
  * sleeve generation
  * portfolio construction
  * comparison and unified review

This will validate that the StratLake system supports **learned models in addition to deterministic alphas**.

---

### Alpha Implementation

* Create a new alpha:

  ```plaintext
  ml_cross_sectional_xgb_2026_q1
  ```
* Implement in:

  ```plaintext
  src/research/alphas/ml_cross_sectional_xgb.py
  ```
* Use gradient boosting (XGBoost or LightGBM)
* Cross-sectional regression on `features_daily`
* Target: `target_ret_5d`

---

### Training Design

* Train window:

  ```plaintext
  [2026-01-01, 2026-03-02)
  ```
* Test window:

  ```plaintext
  [2026-03-02, 2026-04-03)
  ```
* Train once, predict once (no rolling retraining)
* Ensure:

  * deterministic ordering
  * fixed random seed
  * no lookahead bias

---

### Artifact Integration

Ensure full compatibility with existing artifact system:

#### Forecast Layer

* `predictions.parquet`
* `alpha_metrics.json`
* `ic_timeseries.csv`
* `promotion_gates.json`

#### Sleeve Layer

* `signals.parquet`
* `signal_mapping.json`
* `sleeve_returns.csv`
* `sleeve_equity_curve.csv`
* `sleeve_metrics.json`

#### Metadata

* `manifest.json` including:

  * model type
  * feature list
  * hyperparameters
  * train/test windows

---

### Signal Mapping

* Use existing mapping:

  ```yaml
  top_bottom_quantile (q=0.2)
  ```
* Persist mapping name deterministically

---

### Portfolio Integration

* Extend:

  ```plaintext
  configs/portfolios_alpha_2026_q1.yml
  ```
* Add ML alpha sleeve as a component (`artifact_type: alpha_sleeve`)
* Validate portfolio construction with ML alpha

---

### Comparison

* Enable comparison via:

  * `forecast` view
  * `sleeve` view
  * `combined` view
* Compare ML alpha vs deterministic baseline

---

### Unified Review

* Ensure ML alpha appears in:

  * unified review
  * with forecast + sleeve context
* Preserve forecast-first ranking semantics

---

### Documentation

Create:

```plaintext
docs/ml_alpha_2026_q1_case_study.md
```

Include:

* model description
* feature inputs
* training setup
* evaluation results
* sleeve performance
* comparison vs baseline
* interpretation

---

## **Acceptance Criteria**

* ML alpha produces deterministic outputs across runs
* Alpha evaluation metrics (IC, ICIR) are computed correctly
* Sleeve artifacts are generated and valid
* Portfolio can be constructed using ML alpha sleeve
* Alpha appears correctly in:

  * comparison CLI
  * unified review
* No regression in existing deterministic alpha workflows
* Documentation clearly explains results and tradeoffs

---

## **Suggested Release Content**

* First ML-based alpha integrated into StratLake pipeline
* Demonstrates compatibility of learned models with deterministic research framework
* Enables direct comparison between rule-based and ML alphas
* Extends system toward production-grade multi-alpha research

---

## **Notes**

* Prioritize determinism and reproducibility over model complexity
* Do not introduce hyperparameter tuning or ensembling in this milestone
* Ensure strict separation between:

  * forecast quality
  * sleeve performance
* This is a **validation case study**, not a production alpha

---

## **Outcome**

A fully integrated ML alpha that:

* flows through the entire StratLake pipeline
* produces comparable forecast and sleeve metrics
* demonstrates how machine learning models fit into the existing architecture
* enables side-by-side evaluation with deterministic alphas
